### PR TITLE
VideoPlayer: drop FlipPage from RMs public interface

### PR DESCRIPTION
**Pull request by FernetMenta**

### CODE REVIEW

**Comment by FernetMenta** [comment]
@popcornmix I think this requires adaptions in omx player

@garbear not sure if you have correct pts in videopicture. please check

File: xbmc/cores/RetroPlayer/RetroPlayerVideo.cpp [hunk 1]
```diff
@@ -208,14 +208,10 @@ void CRetroPlayerVideo::SendPicture(VideoPicture& picture)
 {
   std::atomic_bool bAbortOutput(false); //! @todo
 
-  int index = m_renderManager.AddVideoPicture(picture);
+  if (!m_renderManager.AddVideoPicture(picture, bAbortOutput, VS_INTERLACEMETHOD_NONE, false))
   if (index < 0)
```
Comment (popcornmix) [inline xbmc/cores/RetroPlayer/RetroPlayerVideo.cpp]: This doesn't compile - no index defined any more.

Comment (FernetMenta) [inline xbmc/cores/RetroPlayer/RetroPlayerVideo.cpp]: updated

**Comment by popcornmix** [comment]
@FernetMenta any advice on what is required by omxplayer now?

I used to call Flip (but not AddVideoPicture) to update pts value in renderer.

Now I don't really want to call AddVideoPicture as there is no picture to display.

If I don't call either, then I don't get any subs rendered.

**Comment by FernetMenta** [comment]
@popcornmix actually nothing has changed. Both methods were called in a sequence. pts value argument of Flip was same as picture->pts of AddVideopicture

**Comment by FernetMenta** [comment]
@popcornmix how is the status here?

**Comment by popcornmix** [comment]
omxplayer has never called AddVideoPicture. It has no pictures to add - video rendering occurs on the gpu with the arm having no knowledge of individual frames being renderered.

omplayer does call FlipPage:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/omxplayer/OMXPlayerVideo.cpp#L313
which is required for subtitles to work.

I'm not sure what is now required. Should I generate a fake picture to pass to AddVideoPicture that gets discarded by renderer? It seems inefficient.

**Comment by FernetMenta** [comment]
Call AddVIdeoPicture with a picture that has required parameters. You renderer can ignore the picture or whatever is appropriate for it.
At a later stage Renderer::FipPage will die as well becuase it is obsolete with buffering in renderer.

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**